assertions/screenshotIdenticalToBaseline.js

```diff
@@ -16,19 +16,23 @@ const compareWithBaseline = require('../lib/compare-with-baseline')
  * that order. Further assertions will compare against the screenshot that was
  * saved in the first execution of the assertion.
  *
- * @param {String} selector Identifies the element that will be captured in the screenshot.
- * @param {String} fileName Optional file name for this screenshot; defaults to the selector
+ * @param elementId Identifies the element that will be captured in the screenshot. Could be string or page object.
+ * @param {String} fileName Optional file name for this screenshot; defaults to the elementId
  * @param {NightwatchVRTOptions} settings Optional settings to override the defaults and `visual_regression_settings`
  * @param {String} message Optional message for `nightwatch` to log upon completion
  */
 exports.assertion = function screenshotIdenticalToBaseline(
-    elementId = 'body',
+    elementId,
     fileName = elementId,
     settings,
     message
 ) {
-
-    this.message = message || `Visual regression test results for element <${elementId.selector}>.`
+    if (elementId.selector) {
```

Comment:
I will better try to look for the key like if (elementId.hasOwnProperty('selector')){console.log(elementId.selector)}

Reply:
If you don't have key i.e selector it will return `undefined` not boolean

Reply:
is undefined not falsey?

Reply:
Have spoken to Vishal and walked him through the tests I've run locally; the suggested change actually breaks things for the worse here. Seems that it may not be a true object being bought through, but the code as is, does what is intended